static/js/ui_init.js

```diff
@@ -74,10 +74,8 @@ exports.initialize_kitchen_sink_stuff = function () {
             if (message_viewport.at_top()) {
                 navigate.up();
             }
-        } else if (delta > 0) {
-            if (message_viewport.at_bottom()) {
-                navigate.down();
-            }
+        } else if (delta > 0 && message_viewport.at_bottom()) {
+            navigate.down();
         }
```

Comment:
I feel like this code is a bit worse off due to this rule; but I think probably the fix would be to remove the `else` bit and have both conditions apply this rule.

Reply:
#16952